packages/indexer-agent/src/network.ts

```diff
@@ -435,28 +446,167 @@ export class Network {
         throw result.error
       }
 
-      return result.data.allocations.map(
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (allocation: any) => ({
-          id: allocation.id,
-          subgraphDeployment: {
-            id: new SubgraphDeploymentID(allocation.subgraphDeployment.id),
-            stakedTokens: BigNumber.from(
-              allocation.subgraphDeployment.stakedTokens,
-            ),
-            signalAmount: BigNumber.from(
-              allocation.subgraphDeployment.signalAmount,
-            ),
+      return result.data.allocations.map(parseGraphQLAllocation)
+    } catch (error) {
+      const err = indexerError(IndexerErrorCode.IE011, error)
+      this.logger.error(INDEXER_ERROR_MESSAGES[IndexerErrorCode.IE011], {
+        err,
+      })
+      throw err
+    }
+  }
+
+  async disputableAllocations(
+    currentEpoch: BigNumber,
+    deployments: SubgraphDeploymentID[],
+    minimumAllocation: number,
+  ): Promise<Allocation[]> {
+    let dataRemaining = true
+    let allocations: Allocation[] = []
+
+    try {
+      const zeroPOI = utils.hexlify(Array(32).fill(0))
+      // TODO: Use disputable epochs period value from the network instead of hardcoding to 2 epochs
+      const disputableEpoch = currentEpoch.toNumber() - 2
```

Comment:
Interesting. We can set a constant for this, or even better to read from a config variable. Disputable allocations will depend very much on a social contract called the Arbitration Charter, probably we want let indexers to change this number easily.

Reply:
I've added a startup parameter to configure this constant, `--poi-disputable-epochs`.